src/interactions/panZoomInteraction.ts

```diff
@@ -20,56 +29,115 @@ export module Interaction {
-        this._xScale = xScale;
-        // HACKHACK #1388: self-register for resetZoom()
-        this._xScale.broadcaster.registerListener("pziX" + this.getID(), () => this.resetZoom());
-      }
-      if (yScale) {
-        this._yScale = yScale;
-        // HACKHACK #1388: self-register for resetZoom()
-        this._yScale.broadcaster.registerListener("pziY" + this.getID(), () => this.resetZoom());
-      }
-    }
+      this._xScale = xScale;
+      this._yScale = yScale;
 
-    /**
-     * Sets the scales back to their original domains.
-     */
-    public resetZoom() {
-      // HACKHACK #254
-      this._zoom = d3.behavior.zoom();
-      if (this._xScale) {
-        this._zoom.x((<any> this._xScale)._d3Scale);
-      }
-      if (this._yScale) {
-        this._zoom.y((<any> this._yScale)._d3Scale);
-      }
-      this._zoom.on("zoom", () => this._rerenderZoomed());
-      this._zoom(this._hitBox);
+      this._dragInteraction = new Interaction.Drag();
+      this._setupDragInteraction();
+      this._touchIds = d3.map();
     }
 
     public _anchor(component: Component.AbstractComponent, hitBox: D3.Selection) {
       super._anchor(component, hitBox);
-      this.resetZoom();
+      this._dragInteraction._anchor(component, hitBox);
+
+      var mouseDispatcher = Dispatcher.Mouse.getDispatcher(<SVGElement> this._componentToListenTo.content().node());
+      mouseDispatcher.onWheel("Interaction.PanZoom" + this.getID(), (p: Point, e: WheelEvent) => this._handleWheelEvent(p, e));
+
+      this._touchDispatcher = Dispatcher.Touch.getDispatcher(<SVGElement> this._componentToListenTo.content().node());
+      this._touchDispatcher.onTouchStart("Interaction.PanZoom" + this.getID(),
+        (ids, idToPoint, e) => this._handleTouchStart(ids, idToPoint, e));
+      this._touchDispatcher.onTouchMove("Interaction.PanZoom" + this.getID(),
+        (ids, idToPoint, e) => this._handleTouchMove(ids, idToPoint, e));
+      this._touchDispatcher.onTouchEnd("Interaction.PanZoom" + this.getID(),
+        (ids, idToPoint, e) => this._handleTouchEnd(ids, idToPoint, e));
     }
 
-    public _requiresHitbox() {
-      return true;
+    private _handleTouchStart(ids: number[], idToPoint: { [id: number]: Point; }, e: TouchEvent) {
+      ids.forEach((id) => {
+        if (this._touchIds.size() === 2) {
+          return;
+        }
+
+        this._touchIds.set(id.toString(), this._translateToComponentSpace(idToPoint[id]));
+      });
     }
 
-    private _rerenderZoomed() {
-      // HACKHACK since the d3.zoom.x modifies d3 scales and not our TS scales, and the TS scales have the
-      // event listener machinery, let's grab the domain out of the d3 scale and pipe it back into the TS scale
-      if (this._xScale) {
-        var xDomain = (<any> this._xScale)._d3Scale.domain();
-        this._xScale.domain(xDomain);
+    private _handleTouchMove(ids: number[], idToPoint: { [id: number]: Point; }, e: TouchEvent) {
+      if (this._touchIds.size() !== 2) {
```

Comment:
Any extra Touches beyond the second one are completely ignored, unless you want different functionality.